dotnetv3/SageMaker/Scenarios/PipelineWorkflow.cs

```diff
@@ -0,0 +1,738 @@
+
+        Console.WriteLine(new string('-', 80));
+    }
+
+    /// <summary>
+    /// Set up the bucket to use for pipeline input and output.
+    /// </summary>
+    /// <param name="bucketName">The name for the bucket.</param>
+    /// <returns>Async task.</returns>
+    public static async Task SetupBucket(string bucketName)
+    {
+        Console.WriteLine(new string('-', 80));
+        Console.WriteLine($"Setting up bucket {bucketName}.");
+
+        var bucketExists = await Amazon.S3.Util.AmazonS3Util.DoesS3BucketExistV2Async(_s3Client,
+            bucketName);
+
+        if (!bucketExists)
+        {
+            await _s3Client.PutBucketAsync(new PutBucketRequest()
+            {
+                BucketName = bucketName,
+                BucketRegion = S3Region.USWest2
+            });
+
+            Thread.Sleep(5000);
+
+            await _s3Client.PutObjectAsync(new PutObjectRequest()
+            {
+                BucketName = bucketName,
+                Key = "samplefiles/latlongtest.csv",
+                FilePath = "latlongtest.csv"
+            });
+        }
+
+        Console.WriteLine($"\tBucket {bucketName} ready.");
+        Console.WriteLine(new string('-', 80));
+    }
+
+    /// <summary>
+    /// Display some results from the output directory.
+    /// </summary>
+    /// <param name="bucketName">The name for the bucket.</param>
+    /// <returns>Async task.</returns>
+    public static async Task<string> GetOutputResults(string bucketName)
+    {
+        Console.WriteLine(new string('-', 80));
+        Console.WriteLine($"Getting output results {bucketName}.");
+        string outputKey = "";
+        Thread.Sleep(15000);
+        var outputFiles = await _s3Client.ListObjectsAsync(
+              new ListObjectsRequest()
+              {
+                  BucketName = bucketName,
+                  Prefix = "outputfiles/"
+              });
+
+        if (outputFiles.S3Objects.Any())
+        {
+            var sampleOutput = outputFiles.S3Objects.OrderBy(s => s.LastModified).Last();
+            Console.WriteLine($"\tOutput file: {sampleOutput.Key}");
+            var outputSampleResponse = await _s3Client.GetObjectAsync(
+                new GetObjectRequest()
+                {
+                    BucketName = bucketName,
+                    Key = sampleOutput.Key
+                });
+            outputKey = sampleOutput.Key;
+            StreamReader reader = new StreamReader(outputSampleResponse.ResponseStream);
+            await reader.ReadLineAsync();
+            Console.WriteLine("\tOutput file contents: \n");
+            for (int i = 0; i < 10; i++)
+            {
+                if (!reader.EndOfStream)
+                {
+                    Console.WriteLine("\t" + await reader.ReadLineAsync());
+                }
+            }
+        }
+
+        Console.WriteLine(new string('-', 80));
+        return outputKey;
+    }
+
+    /// <summary>
+    /// Create a pipeline from some json.
```

Comment:
I feel like this summary should be more specific.

Reply:
Updated with some more details.